immersion/kubeflow_pipelines/pipelines/solutions/pipeline_vertex/pipeline.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+# file except in compliance with the License. You may obtain a copy of the License at
+
+# https://www.apache.org/licenses/LICENSE-2.0
+    
+# Unless required by applicable law or agreed to in writing, software 
+# distributed under the License is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either 
+# express or implied. See the License for the specific language governing 
+# permissions and limitations under the License.
+"""Kubeflow Covertype Pipeline."""
+import os
+
+from kfp import dsl
+from kfp.components import create_component_from_func_v2
+
+from tuning_lightweight_component import tune_hyperparameters
+from training_lightweight_component import train_and_deploy
+
+
+PIPELINE_ROOT = os.getenv('PIPELINE_ROOT')
+PROJECT_ID = os.getenv('PROJECT_ID')
+REGION = os.getenv('REGION')
+
+TRAINING_CONTAINER_IMAGE_URI = os.getenv('TRAINING_CONTAINER_IMAGE_URI')
+SERVING_CONTAINER_IMAGE_URI = os.getenv('SERVING_CONTAINER_IMAGE_URI')
+
+TRAINING_FILE_PATH = os.getenv('TRAINING_FILE_PATH')
+VALIDATION_FILE_PATH = os.getenv('VALIDATION_FILE_PATH')
+
+MAX_TRIAL_COUNT = os.getenv('MAX_TRIAL_COUNT', 5)
+PARALLEL_TRIAL_COUNT = os.getenv('PARALLEL_TRIAL_COUNT', 5)
+THRESHOLD = os.getenv('THRESHOLD', 0.6)
+
+
+tune_hpyerparameters_component = create_component_from_func_v2(
+    tune_hyperparameters,
+    base_image='python:3.8',
```

Comment:
nit: the current GCP Notebooks default to Python 3.7, so that would be the best Py version to use, but I do not expect this to cause any issue so can stay like this to future-proof